src/verification_key.rs

```diff
@@ -148,44 +134,32 @@ impl VerificationKey {
     ///
     /// ## Zcash-specific consensus properties
     ///
-    /// Ed25519 checks are described in [§5.4.5][ps] of the Zcash protocol
-    /// specification. Ed25519 validation is not well-specified, and the original
-    /// implementation of JoinSplit signatures for `zcashd` inherited its precise
-    /// validation rules from a specific build configuration of `libsodium`
-    /// `1.0.15`. Note that the precise version is important because `libsodium`
-    /// changed validation rules in point releases.
+    /// Ed25519 checks are described in [§5.4.5][ps] of the Zcash protocol specification and in
+    /// [ZIP 215].  The verification criteria for an (encoded) signature `(R_bytes, s_bytes)` with
+    /// (encoded) verification key `A_bytes` are:
     ///
-    /// The additional validation checks are that:
+    /// * `A_bytes` and `R_bytes` MUST be encodings of points `A` and `R` respectively on the
+    ///   twisted Edwards form of Curve25519, and non-canonical encodings MUST be accepted;
     ///
-    /// * `s` MUST represent an integer less than the prime `l`, per `libsodium`
-    /// `1.0.15` `crypto_sign/ed25519/ref10/open.c:126`;
+    /// * `s_bytes` MUST represent an integer `s` less than `l`, the order of the prime-order
+    ///   subgroup of Curve25519;
     ///
-    /// * `R` MUST NOT be one of the excluded encodings, per `libsodium` `1.0.15`
-    /// `crypto_sign/ed25519/ref10/open.c:127`;
+    /// * the verification equation `[8][s]B = [8]R + [8][k]A` MUST be satisfied;
     ///
-    /// * The public key bytes must not be all 0, per `libsodium` `1.0.15`
-    /// `crypto_sign/ed25519/ref10/open.c:138-143`, which we maintain as an
-    /// invariant on the `VerificationKey` type.
+    /// * the alternate verification equation `[s]B = R + [k]A`, allowed by RFC 8032, MUST NOT be
+    ///   used.
     ///
-    /// [ps]: https://zips.z.cash/protocol/protocol.pdf#concretejssig
+    /// [ps]: https://zips.z.cash/protocol/protocol.pdf#concretejssig [ZIP 215]:
```

Comment:
This moved to https://zips.z.cash/protocol/protocol.pdf#concreteed25519

Reply:
:+1: